src/test/java/org/testng/internal/paramhandler/FakeTestNGMethod.java

```diff
@@ -0,0 +1,379 @@
+package org.testng.internal.paramhandler;
+
+import org.testng.IClass;
+import org.testng.IRetryAnalyzer;
+import org.testng.ITestClass;
+import org.testng.ITestNGMethod;
+import org.testng.internal.ConstructorOrMethod;
+import org.testng.internal.MethodInstanceTest;
+import org.testng.internal.reflect.ReflectionHelper;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlTest;
+
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+public class FakeTestNGMethod implements ITestNGMethod {
+    private ConstructorOrMethod consMethod;
+    private Class<?> clazz;
+    private XmlTest xmlTest;
+
+    public FakeTestNGMethod(Class<?> clazz, String methodName) {
+        this.clazz = clazz;
+        Method[] methods = ReflectionHelper.getLocalMethods(clazz);
+        for (Method method : methods) {
+            if (method.getName().equalsIgnoreCase(methodName)) {
+                consMethod = new ConstructorOrMethod(method);
+                break;
+            }
+        }
+    }
+
+    public FakeTestNGMethod(Class<?> clazz, String methodName, XmlTest xmlTest) {
+        this(clazz, methodName);
```

Comment:
Usually, we do the opposite: the constructor with less parameters call the constructor with more parameters (with default values).

```java
public FakeTestNGMethod(Class<?> clazz, String methodName) {
  this(clazz, methodName, null);
}
```

Reply:
Fixed.